src/core/ddsc/src/dds_reader.c

```diff
@@ -660,13 +669,24 @@ static dds_entity_t dds_create_reader_int (dds_entity_t participant_or_subscribe
 
     iox_sub_options_t opts;
     iox_sub_options_init(&opts);
-
-    // ICEORYX TODO: handle failure (how should the system behave if resources are insufficient?)
+    
     iox_sub_storage_extension_init(&rd->m_iox_sub_stor);
 
     assert (rqos->durability.kind == DDS_DURABILITY_VOLATILE);
-    opts.queueCapacity = rd->m_entity.m_domain->gv.config.sub_queue_capacity;
+    
+    // NB: Conversion is ok as history.depth is > 0
+    // If the value is larger than the max supported by iceoryx this would saturate in the iox_pub_init,
+    // but currently we do not use iceoryx in this case.
+
+    // NB: We may lose data after history.depth many samples are received (if we are not taking them fast enough from iceoryx and move them in the reader history cache), but this is valid behavior
+    // It may still lead to undesired behavior as the queues are filled very fast 
+    // if data is published as fast as possible
+    // Essentially this is another level of caching/buffering data.
+    opts.queueCapacity = (uint64_t)rqos->history.depth;
```

Comment:
min(max_possible, depth) (depth)
depends on reliability as well when we should driop data

for keyed topics this may no be enough to guarantee delivery up to a certain history depth but this cannot be solved with the iceoryx queue model. 

Assume we have a topic with two keys, to guarantee delivery of up to n elements for each key it does not suffice to have a queue of size 2n since we cannot control that iceoryx pushes samples into the queue for each key fairly/evenly.